python/neuroglancer/downsample.py

```diff
@@ -23,9 +23,21 @@ def method(layer_type):
     return downsample_with_averaging
   elif layer_type == 'segmentation':
     return downsample_segmentation
+  elif layer_type == 'activation':
+    return downsample_with_max_pooling
   else:
     return downsample_with_striding 
 
+def validate_factor(array, factor):
+  factor = np.array(factor, dtype=np.int32)
+  if np.any(factor <= 0):
+    raise ValueError("Factors less than one don't make sense. Factor: {}".format(factor))
+
+  if len(array.shape) == 4 and len(factor) == 3:
```

Comment:
This is mainly so if you're like oh i want to downsample this 3D image, but it has lots of channels (e.g. RGB), it'll do what you mean.